Add codespell support (config, workflow to detect/not fix) and make it fix some typos

yarikoptic opened the pull request:
More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

There might still be a few false positives, hopefully tests on CI would catch them ;)

yarikoptic committed:
Add github action to codespell develop on push and PRs

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
Tune up excludes for codespell

yarikoptic committed:
[DATALAD RUNCMD] chore: run codespell throughout fixing a few typos interactively

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w -i 3 -C 4",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

MilosPaunovic commented on ui/src/components/dependencies/composables/useDependencies.ts:
```diff
@@ -223,7 +223,7 @@ function selectHandler(cy: cytoscape.Core, node: cytoscape.NodeSingular, selecte
 
     // Smoothly center and zoom the viewport on the selected node
-    cy.animate({center: {eles: node}, zoom: 1.2}, {duration: 500});
+    cy.animate({center: {else: node}, zoom: 1.2}, {duration: 500});
```
This is not good, it's not a spelling error, it's a name of the property of an external library.

yarikoptic replied:
coolio -- would you prefer me to add inline pragma to ignore just here or ignore this typo (`eles`) generally?

MilosPaunovic replied:
Is it possible to add it to ignore in just this particular file?

yarikoptic replied:
sure -- proposed above in config  

https://github.com/kestra-io/kestra/pull/12912/files#r2557747759

MilosPaunovic replied:
This change would have to reverted.

MilosPaunovic reviewed (changes requested):
For the 4 `UI` related files, it has done a solid job except one mistake of handling a property name, but I'm afraid it will not be okay from the `BE` files perspective.

MilosPaunovic commented:
I've taken the changes related to `UI` components and updated them here https://github.com/kestra-io/kestra/pull/12928.

MilosPaunovic commented:
Hey @loicmathieu, can you check if this makes sense once you get the chance?

yarikoptic commented on .codespellrc:
```diff
@@ -0,0 +1,8 @@
+[codespell]
+# Ref: https://github.com/codespell-project/codespell#using-a-config-file
+skip = .git*,*.svg,package-lock.json,*.css,.codespellrc,translations,*.patch
```
```suggestion
skip = .git*,*.svg,package-lock.json,*.css,.codespellrc,translations,*.patch,useDependencies.ts
```
ref - below https://github.com/kestra-io/kestra/pull/12912/files#r2532935221

loicmathieu commented on core/src/test/java/io/kestra/core/runners/pebble/filters/EscapeCharFilterTest.java:
please revert

yarikoptic replied:
Right! Sorry missed that... Can ignore this entire file or add those inline ignores. What do you prefer?

loicmathieu replied:
Inline ignores would be better

loicmathieu reviewed (changes requested):
Overall I'm not against it but we would need first to discuss this internally to see if we agree to add it or not.

Is there some way to prevent codespell inline in the code, like special comments?

yarikoptic commented:
Sure thing - definitely don't follow blindly! 

> Is there some way to prevent codespell inline in the code, like special comments?

Yes
https://github.com/codespell-project/codespell?tab=readme-ov-file#inline-ignore

loicmathieu commented:
@yarikoptic we're ready to merge this PR soon, can you rebase it and made the requested changes?

yarikoptic committed:
Add inline codespell ignores in a few spots

yarikoptic commented:
done!

MilosPaunovic commented on ui/src/components/dependencies/composables/useDependencies.ts:
```diff
@@ -223,7 +223,7 @@ function selectHandler(cy: cytoscape.Core, node: cytoscape.NodeSingular, selecte
 
     // Smoothly center and zoom the viewport on the selected node
-    cy.animate({center: {eles: node}, zoom: 1.2}, {duration: 500});
+    cy.animate({center: {else: node}, zoom: 1.2}, {duration: 500});
```
This is plain wrong, needs to excluded.

loicmathieu replied:
Please use inline comment to ignore

loicmathieu commented on core/src/test/java/io/kestra/plugin/core/namespace/PurgeFilesTest.java:
```diff
@@ -82,7 +82,7 @@ void should_find_all_namespaces_with_glob_pattern() throws IllegalVariableEvalua
 
         PurgeFiles purgeFiles = PurgeFiles.builder()
             .type(PurgeFiles.class.getName())
-            .namespacePattern(Property.ofValue("*arent*"))
```
Please revert, this is a pattern for `parents`

loicmathieu replied:
Please use inline comment to ignore

loicmathieu commented on .codespellrc:
```diff
@@ -0,0 +1,8 @@
+skip = .git*,*.svg,package-lock.json,*.css,.codespellrc,translations,*.patch,data
+check-hidden = true
+# mixed case, latin
+ignore-regex = \b([a-zA-Z]+[A-Z]|[A-Z][a-z])[a-zA-Z_-]*\b|L.eau c.est la vie
```
Why this regex?
Anyway the last part of the pattern should be removed and replaced by an inline ignore comment.

yarikoptic replied:
comment above tried to explain the regex for mixed case (camelCase and PascalCase like), without it we would get hits in "unfortunate" to be close to typo interfaces which we should not (`notIn`) or likely do not want (`Searcheable`) to change at this point

<details>
<summary>full list </summary> 

```shell
❯ codespell
./cli/src/test/java/io/kestra/cli/services/FileChangedEventListenerTest.java:45: AfterAll ==> after all
./core/src/main/java/io/kestra/core/models/QueryFilter.java:66: NotIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/models/dashboards/filters/AbstractFilter.java:28: NotIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/models/dashboards/filters/NotIn.java:21: NotIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/runners/pebble/AbstractIndent.java:69: Unknow ==> Unknown
./core/src/main/java/io/kestra/core/runners/pebble/functions/AbstractFileFunction.java:178: customNs ==> customs
./core/src/main/java/io/kestra/core/runners/pebble/functions/AbstractFileFunction.java:179: customNs ==> customs
./core/src/main/java/io/kestra/core/runners/pebble/functions/AbstractFileFunction.java:180: customNs ==> customs
./core/src/main/java/io/kestra/core/runners/pebble/functions/AbstractFileFunction.java:182: customNs ==> customs
./core/src/main/java/io/kestra/core/services/AbstractFilterService.java:32: NotIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/services/AbstractFilterService.java:32: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/services/AbstractFilterService.java:70: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/services/AbstractFilterService.java:70: NotIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/test/flow/Assertion.java:49: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/test/flow/Assertion.java:56: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/test/flow/Assertion.java:71: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/test/flow/Assertion.java:163: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/core/test/flow/Assertion.java:364: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/plugin/core/condition/ExecutionStatus.java:63: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/plugin/core/condition/ExecutionStatus.java:79: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/plugin/core/condition/HasRetryAttempt.java:64: notIn ==> noting, not in, notion
./core/src/main/java/io/kestra/plugin/core/condition/HasRetryAttempt.java:74: notIn ==> noting, not in, notion
./core/src/test/java/io/kestra/core/test/AssertionTest.java:316: notIn ==> noting, not in, notion
./core/src/test/java/io/kestra/core/test/AssertionTest.java:326: notIn ==> noting, not in, notion
./core/src/test/java/io/kestra/core/utils/NamespaceFilesUtilsTest.java:77: customNs ==> customs
./core/src/test/java/io/kestra/plugin/core/condition/ExecutionStatusTest.java:39: notIn ==> noting, not in, notion
./core/src/test/java/io/kestra/plugin/core/condition/ExecutionStatusTest.java:44: notIn ==> noting, not in, notion
./core/src/test/java/io/kestra/plugin/core/condition/ExecutionStatusTest.java:59: notIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/repository/AbstractJdbcExecutionRepository.java:1079: NotIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/repository/AbstractJdbcRepository.java:345: notIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/repository/AbstractJdbcServiceInstanceRepository.java:152: notIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/services/JdbcFilterService.java:46: NotIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/services/JdbcFilterService.java:119: notIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/services/JdbcFilterService.java:119: NotIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/services/JdbcFilterService.java:120: notIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/services/JdbcFilterService.java:194: NotIn ==> noting, not in, notion
./jdbc/src/main/java/io/kestra/jdbc/services/JdbcFilterService.java:195: notIn ==> noting, not in, notion
./ui/.jshintrc:27: afterAll ==> after all
./ui/src/components/logs/linkify.ts:20: Tranform ==> Transform
./webserver/src/main/java/io/kestra/webserver/controllers/api/NamespaceSecretController.java:10: Searcheable ==> Searchable
./webserver/src/main/java/io/kestra/webserver/controllers/api/NamespaceSecretController.java:61: Searcheable ==> Searchable
./webserver/src/main/java/io/kestra/webserver/controllers/api/NamespaceSecretController.java:62: Searcheable ==> Searchable
./webserver/src/main/java/io/kestra/webserver/utils/Searcheable.java:16: Searcheable ==> Searchable
./webserver/src/main/java/io/kestra/webserver/utils/Searcheable.java:20: Searcheable ==> Searchable
./webserver/src/main/java/io/kestra/webserver/utils/Searcheable.java:24: Searcheable ==> Searchable
./webserver/src/main/java/io/kestra/webserver/utils/Searcheable.java:25: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/controllers/api/ExecutionControllerRunnerTest.java:2361: atMost ==> at most, almost, utmost
./webserver/src/test/java/io/kestra/webserver/controllers/api/ExecutionControllerRunnerTest.java:2371: atMost ==> at most, almost, utmost
./webserver/src/test/java/io/kestra/webserver/controllers/api/ExecutionControllerRunnerTest.java:2381: atMost ==> at most, almost, utmost
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:12: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:22: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:27: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:27: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:39: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:39: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:53: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:53: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:67: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:67: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:82: Searcheable ==> Searchable
./webserver/src/test/java/io/kestra/webserver/utils/SearcheableTest.java:82: Searcheable ==> Searchable
```
</details>


<details>
<summary>although it was masking away two typos I now fixed in a dedicated commit </summary> 

and indeed that latin skip is already inlined, so I dropped it now.

```shell
commit 62bd545057303895f69c3fdb35dc0fa448a15b60
Author: Yaroslav Halchenko <debian@onerussian.com>
Date:   Thu Jan 8 10:25:51 2026 -0500

    [DATALAD RUNCMD] chore: run codespell on two files ignoring our mixed case skips
    
    === Do not change lines below ===
    {
     "chain": [],
     "cmd": "codespell -w core/src/main/java/io/kestra/core/runners/pebble/AbstractIndent.java ui/src/components/logs/linkify.ts",
     "exit": 0,
     "extra_inputs": [],
     "inputs": [],
     "outputs": [
      "core",
      "ui"
     ],
     "pwd": "."
    }
    ^^^ Do not change lines above ^^^

diff --git a/core/src/main/java/io/kestra/core/runners/pebble/AbstractIndent.java b/core/src/main/java/io/kestra/core/runners/pebble/AbstractIndent.java
index 8c3dd98d4..ac913c2ce 100644
--- a/core/src/main/java/io/kestra/core/runners/pebble/AbstractIndent.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/AbstractIndent.java
@@ -66,7 +66,7 @@ public abstract class AbstractIndent {
             // nindent filter adds a newline to the string and indents each line by defined amount of spaces
             return (newLine + input).replace(newLine, newLine + prefix.repeat(amount));
         }
-        throw new PebbleException(null, String.format("Unknow indent type '%s'.", indentType), lineNumber, self.getName());
+        throw new PebbleException(null, String.format("Unknown indent type '%s'.", indentType), lineNumber, self.getName());
 
     }
 }
diff --git a/ui/src/components/logs/linkify.ts b/ui/src/components/logs/linkify.ts
index 293c1ffd9..75357eca8 100644
--- a/ui/src/components/logs/linkify.ts
+++ b/ui/src/components/logs/linkify.ts
@@ -17,7 +17,7 @@ function gotoRoute(event: MouseEvent, route: any, router: Router) {
 }
 
 /**
- * Tranform router-md elements in an html element use 
+ * Transform router-md elements in an html element use 
  * the vue-router methods instead of natural links 
  * @param element 
  */
@@ -61,4 +61,4 @@ export default function linkify(element?: HTMLElement, router?: Router) {
     anchorNodeExection.after(anchorNodeFlow);
     anchorNodeExection.after(document.createTextNode(" for flow "));
   });
-}
\ No newline at end of file
+}

```
</details>

yarikoptic replied:
here would be the summary of those mixed case ones -- we could exclude them explicitly here and overall test all the others
```bash
❯ ~/bin/codespellit list-hits-sorted
      1 afterAll ==> after all
      1 AfterAll ==> after all
      3 atMost ==> at most, almost, utmost
      5 customNs ==> customs
      9 NotIn ==> noting, not in, notion
     19 Searcheable ==> Searchable
     21 notIn ==> noting, not in, notion
```

yarikoptic committed:
[DATALAD RUNCMD] chore: run codespell on two files ignoring our mixed case skips

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w core/src/main/java/io/kestra/core/runners/pebble/AbstractIndent.java ui/src/components/logs/linkify.ts",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [
  "core",
  "ui"
 ],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
[DATALAD RUNCMD] chore: run codespell throughout fixing a few  new typos automagically

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic commented:
done! while reviewing, fixed 2 more typos ;)

MilosPaunovic reviewed (approved):
LGTM for the UI related files.

loicmathieu reviewed (approved):
LGTM, thanks a lot!

loicmathieu committed:
Merge branch 'develop' into enh-codespell

loicmathieu commented:
@yarikoptic I rebase this branch and will merge it if everything is OK, thanks a lot for your work, it is a very nice addition.
If it goes well, we plan to add this to all our other repositories, we have more than 100 plugin repositories that would benefit for it ;)

yarikoptic commented:
> If it goes well, we plan to add this to all our other repositories, we have more than 100 plugin repositories that would benefit for it ;)

heh heh... I am yet to find a piece of mind when getting into proliferation of repos in our world too... yet to become proficient with https://github.com/copier-org/copier which is largely promoted to help maintain scaffolding for projects.
